readme.md

```diff
@@ -0,0 +1,132 @@
+Once you've filled your shopping cart with the books you desire, our application provides a streamlined order management system. You can review your orders and add them.
+
+#### 3. Book management with pagination and sorting
+We understand the importance of efficient book discovery. Our application offers a sophisticated search functionality that allows users to find all books by categories, author, title, or by id. With pagination support, you can explore a vast catalog of books without overwhelming your search results.
+
+#### 4. Security and JWT Tokens for User auth
+Security is paramount in our application. We've implemented Spring Security to safeguard your data and transactions. JWT (JSON Web Tokens) are used for secure authentication and authorization, ensuring that only authorized users can access sensitive functionalities.
+
+###  Technologies Used
+Our application leverages several cutting-edge technologies, including:
+- **Spring Data JPA:** For efficient data access and persistence.
+- **Spring Security:** To secure the application and user data.
+- **MapStruct:** For simplified and efficient mapping between DTOs and entities.
+- **Docker:** Containerization for easy deployment and scalability.
+- **OpenAPI**: For documenting our RESTful APIs, providing clear and detailed information about available resources, request payloads, and response formats.
+  **Liquibase**: Used to manage and version the database schema, allowing for easy database schema changes and ensuring data schema consistency across different environments.
+
+
+### Api Functionalities
+In our Online Book Store project, we've diligently followed REST (Representational State Transfer) principles in designing our controllers. Controllers are essential components responsible for handling HTTP requests and responses.
+- **BookController**: Handles book-related operations, including creation, retrieval, update, and deletion of books.
+- **CategoryController**
+- **AuthenticationController**
+- **ShoppingCartController**
+- **OrderController** <br>
+In detail order about functions of controllers you can read at [Features](#features)
+## Getting Started
+
+To get started with our Online Book Store, you can follow the installation instruction provided in the [README](#) file in the project repository. This guide will walk you through setting up the application, configuring the necessary settings, and launching it in your preferred environment.
+Thank you for choosing our Online Book Store project. We hope you enjoy the world of books at your fingertips, and we're committed to providing you with a safe, efficient, and delightful shopping experience. If you have any questions or feedback, please don't hesitate to reach out. Happy reading!
+
+## Prerequisites
+
+List the prerequisites required to run your application. Include things like:
+
+- Java 17 (https://www.oracle.com/java/technologies/javase/jdk17-archive-downloads.html)
+- Maven (https://maven.apache.org/download.cgi)
```

Comment:
there is a wrapper in your project, so it is not required actually, just change step 2 of installation to user mwnw command